### PR TITLE
fix #269383: flip doesn't work for selected noteheads and stems

### PR DESCRIPTION
**Pull request by anatoly-os**
Collect all the beams and chords need to be flipped according to selection and then flip each unique chord and beam.

### CODE REVIEW

**Comment by anatoly-os** [comment]
Well, the commit fixes flipping beams if noteheads or stems have been selected.
BTW, I see in 2.1 that slurs, ties, articulations signs, tuplets are also flip if 'parent' notehead or stem has been flipped. It doesn't work in 3.0. Should I fix it in this PR or create separate issue with one more PR?

PS. In 2.1, if I flip a slur, tie, articulation sign or tuplet directly, it can no more be flipped together with 'parent' notehead or stem by flipping the last. Is it a feature or a bug?

**Comment by anatoly-os** [comment]
Also, in 2.1 there is a bug: if I select odd number of noteheads and stems under the same beam, the beam and the staff will be flipped. If I select even number of ones, the beam will never be flipped.
I see the reason in code - flipping is performed for each selected element. I've fixed it for chords and beams and likely will fix for slurs, ties, articulations signs and tuplets. I believe it should be fixed in 2.2, too.

**Comment by lasconic** [comment]
*PS. In 2.1, if I flip a slur, tie, articulation sign or tuplet directly, it can no more be flipped together with 'parent' notehead or stem by flipping the last. Is it a feature or a bug?*
I believe it's a feature. By default the slur is in "auto" mode. If you flip it you hardcode the direction of the slur.

*If I select even number of ones, the beam will never be flipped.*
Feel free to provide a PR on 2.2 branch

**Comment by lasconic** [comment]
@wschweer can you review and eventually close or merge this one ? I see you participated in the discussion here https://musescore.org/en/node/269383

**Comment by anatoly-os** [comment]
I'm preparing more general commit. Will update soon.

**Comment by anatoly-os** [comment]
@wschweer @lasconic done. Commit can be reviewed. Going to fix 2.2 in the same way.

File: libmscore/edit.cpp [hunk 1]
```diff
@@ -1214,35 +1214,53 @@ void Score::cmdFlip()
...
+                  alreadyFlippedElements.insert(element);
+                  flipFunction();
+                  }
+            };
       for (Element* e : el) {
             if (e->isNote() || e->isStem() || e->isHook()) {
-                  Chord* chord;
-                  if (e->isNote())
-                        chord = toNote(e)->chord();
+                  Chord* chord = nullptr;
```
Comment (wschweer) [inline libmscore/edit.cpp]: Initialization is not needed, especially as you do not check for nullptr later on.

**Pull request merged**